migrations/v6.go

```diff
@@ -0,0 +1,36 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+
+	wf_db "github.com/writeas/writefreely/db"
+)
+
+func oauthAttach(db *datastore) error {
+	dialect := wf_db.DialectMySQL
+	if db.driverName == driverSQLite {
+		dialect = wf_db.DialectSQLite
+	}
+	return wf_db.RunTransactionWithOptions(context.Background(), db.DB, &sql.TxOptions{}, func(ctx context.Context, tx *sql.Tx) error {
+		builders := []wf_db.SQLBuilder{
+			dialect.
+				AlterTable("oauth_client_states").
+				AddColumn(dialect.
+					Column(
+						"attach_user_id",
+						wf_db.ColumnTypeInteger,
+						wf_db.OptionalInt{Set: true, Value: 24,}).SetNullable(false).SetDefault("0")),
```

Comment:
Should this column be NULLable for the cases where we're not attaching a user?

Reply:
Personal preference, but I can change it. I avoid null like the plague. I'd rather have default values for primatives whenever possible. I think it translates to Go better and reduces complexity of simple values because you don't have to do "is this is set *and/or* is this a non-default value?" validation.

Reply:
Yeah, it's definitely nicer in Go without them -- they're a pain to work with. But in general, I think it's good database design to put NULLs in optional columns. It conveys the right information in the data itself (e.g. this column was designed to be empty sometimes vs. maybe a bug made this column empty), and helps if we ever need to troubleshoot or to change it somehow. And overall for the database design, I think consistency is best.

Of course, I also think it's fine to just return an `int64` (instead of `sql.NullInt64`) from our database funcs that fetch the value, as you're already doing.